packages/tldraw/src/lib/defaultExternalContentHandlers.ts

```diff
@@ -250,62 +226,89 @@ export function registerDefaultExternalContentHandlers(
 				: editor.getViewportPageBounds().center)
 
 		const pagePoint = new Vec(position.x, position.y)
-
 		const assets: TLAsset[] = []
+		const assetsToUpdate: {
+			asset: TLAsset
+			file: File
+			temporaryAssetPreview?: string
+		}[] = []
+		for (const file of files) {
```

Comment:
looks like we've switched from doing this in parallel to doing them in series. why is that?

Reply:
the parallel part was moved down below! this part doesn't need to be in parallel up here.

however, now that i look at it again, i do notice for high-res images (when adding several at a time), that this hashing makes it feel sluggish: `const hash = await getHashForBuffer(await file.arrayBuffer())`. it's a little bit even worse because we do it twice: once in the `files` and then again in `file` to make them match up.

we can just use the `nanoid` fallback, which seems altogether sufficient, and then pass the assetid when calling `getAssetForExternalContent` right?

is this hash needed anywhere else? i suppose it helps when adding the same asset over & over? but how common is that really and does that really help that much vs. just re-uploading the asset?

here's an example of me adding an 8MB image, _just_ doing the image hashing once (not even the 2nd pass) 😬 :
<img width="418" alt="Screenshot 2024-08-01 at 11 42 07" src="https://github.com/user-attachments/assets/496c1876-d861-4395-8ae2-471189dad96e">

or, at least it's semi-random depending on the browsers load, here's me adding 4 images at a time of similar size:
<img width="388" alt="Screenshot 2024-08-01 at 11 44 18" src="https://github.com/user-attachments/assets/47b2f5cc-da7c-4ea7-b81e-b2228a503080">

Reply:
@SomeHats I went ahead and made this change: https://github.com/tldraw/tldraw/pull/3988/commits/c8ba7f66635266c1b3cf4a39f6dc8800091b4297

lemme know if you disagree!

Reply:
talked offline with @SomeHats , we need to address this but we're going to address it in a separate PR so as to not block this PR anymore!